assets/js/contact.js

```diff
@@ -1,3 +1,16 @@
+async function sha256(message) {
+  const data = new TextEncoder().encode(message);
+  const buffer = await crypto.subtle.digest("SHA-256", data);
+  const array = Array.from(new Uint8Array(buffer));
+  return array
+    .map((b) => b.toString(16).padStart(2, "0"))
+    .join("");
+}
```

Comment:
@aduth would love your feedback here, we had trouble getting this to work locally, the code returns the right results, but we think that because this is async, the `event.preventDefault()` is too late, and doesn't work? Do you have any other ideas for how we could prevent form submission? A quick search didn't turn up any re-implementations of SHA-256 that were synchronous

Reply:
I haven't had time to dive in, but at a glance, maybe we could `setCustomValidity` on the input as it changes, then trust native form validation to prevent submission.

Reply:
Although, based on the code, I would expect that `preventDefault` would be effective in preventing the form submission 🤔

Reply:
I'm not sure if `setCustomValidity` is the way we want to go because we are mocking the submission.